In [4]:
import os
os.chdir('C:\\Users\\Usuario\\Desktop\\wallace\\UNILA\\DNA IC')

In [5]:
data1= open('EIt.txt', 'r')
data2= open('EIf.txt', 'r')
data3= open('IEt.txt', 'r')
data4= open('IEf.txt', 'r')
f1= data1.readlines()
f2= data2.readlines()
f3= data3.readlines()
f4= data4.readlines()
del(f1[0:4])
del(f2[0:4])
del(f3[0:4])
del(f4[0:4])

In [6]:
data=[]
targ=[]
for i in range(len(f1)):
    f1[i]= f1[i].split('): ')
for i in range(len(f1)):
    data.append(f1[i][1])
    targ.append(0)

for i in range(3000):
    f2[i]= f2[i].split(') : ')
for i in range(3000):
    data.append(f2[i][1])
    targ.append(2)

for i in range(len(f3)):
    f3[i]= f3[i].split('): ')
for i in range(len(f3)):
    data.append(f3[i][1])
    targ.append(1)

for i in range(3000):
    f4[i]= f4[i].split(') : ')
for i in range(3000):
    data.append(f4[i][1])
    targ.append(2)


In [7]:
d=[]
for i in range(len(data)):
    da=[]
    for j in range(len(data[i])):
        if data[i][j]=='A':
            da.append(0)
        elif data[i][j]=='C':
            da.append(1)
        elif data[i][j]=='G':
            da.append(2)
        elif data[i][j]=='T':
            da.append(3)
        else:
            da.append(8)
    d.append(da)

In [8]:
from sklearn.feature_selection import SelectFromModel
from sklearn import tree

In [9]:
clf=tree.DecisionTreeClassifier()
clf= clf.fit(d,targ)
model = SelectFromModel(clf, prefit=True, max_features=8)
x = model.transform(d)
print(len(d[0]),len(x[0]))

141 8


In [10]:
data1= open('splice.data.txt', 'r')
f= data1.readlines()
for i in range(len(f)):
    f[i]= f[i].split(',')
for i in range(len(f)):
    del(f[i][1])
for i in range(len(f)):
    f[i][1]=f[i][1].split()[0]
c=[]
for i in range(len(f)):
    b=[]
    if f[i][0]=='EI':
        b.append(0)
    elif f[i][0]=='IE':
        b.append(1)
    elif f[i][0]=='N':
        b.append(2)
    a=[]
    for j in range(len(f[i][1])):
        if f[i][1][j]=='A':
            a.append(0)
        elif f[i][1][j]=='C':
            a.append(1)
        elif f[i][1][j]=='G':
            a.append(2)
        elif f[i][1][j]=='T':
            a.append(3)
        else:
            a.append(8)
    b.append(a)
    c.append(b)
cdata=[]
ctarg=[]

for i in range(len(c)):
    cdata.append(c[i][1])
    ctarg.append((c[i][0]))
clf=tree.DecisionTreeClassifier()
clf= clf.fit(cdata,ctarg)
model = SelectFromModel(clf, prefit=True)
x1 = model.transform(cdata)
print(len(cdata[0]),len(x1[0]))

60 8


In [11]:
import numpy as np
from sklearn import svm
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import ShuffleSplit
from sklearn import tree
from sklearn.svm import SVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import confusion_matrix
from sklearn.metrics import classification_report,confusion_matrix
from sklearn.model_selection import cross_val_predict
from sklearn.linear_model import SGDClassifier
from sklearn.gaussian_process import GaussianProcessClassifier
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import AdaBoostClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.ensemble import VotingClassifier

In [17]:
print('Voting Classifier')
clf1 = GradientBoostingClassifier(n_estimators=100, learning_rate=1.0, max_depth=1, random_state=0)
clf2 = AdaBoostClassifier(n_estimators=100)
clf3 = tree.DecisionTreeClassifier()
clf4 = GaussianNB()
clf5 = SVC()


clf = VotingClassifier(estimators=[('gbc', clf1), ('ada', clf2), ('DTc', clf3), ('gNB', clf4), ('svc', clf5)], voting='hard')
clf.fit(x1, ctarg)
y=clf.predict(x)
cm= confusion_matrix(targ, y)
print(cm)
print("Accuracy: %0.2f, ei: %0.2f , ie: %0.2f, n: %0.2f" % ((cm[0][0]+cm[1][1]+cm[2][2])/sum(cm[0]+cm[1]+cm[2]), cm[0][0]/sum(cm[0]), cm[1][1]/sum(cm[1]),cm[2][2]/sum(cm[2])))
# fit(x1, ctarg) - precisão de 68%

Voting Classifier


C:\Users\Usuario\Anaconda3\lib\site-packages\sklearn\svm\base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


[[2673   87   36]
 [ 116 2707   57]
 [ 953 2488 2559]]
Accuracy: 0.68, ei: 0.96 , ie: 0.94, n: 0.43


In [19]:
print('Voting Classifier')
clf1 = GradientBoostingClassifier(n_estimators=100, learning_rate=1.0, max_depth=1, random_state=0)
clf2 = AdaBoostClassifier(n_estimators=100)
clf3 = tree.DecisionTreeClassifier()
clf4 = GaussianNB()
clf5 = SVC()


clf = VotingClassifier(estimators=[('gbc', clf1), ('ada', clf2), ('DTc', clf3), ('gNB', clf4), ('svc', clf5)], voting='hard')
clf.fit(x, targ)
y=clf.predict(x1)
cm= confusion_matrix(ctarg, y)
print(cm)
print("Accuracy: %0.2f, ei: %0.2f , ie: %0.2f, n: %0.2f" % ((cm[0][0]+cm[1][1]+cm[2][2])/sum(cm[0]+cm[1]+cm[2]), cm[0][0]/sum(cm[0]), cm[1][1]/sum(cm[1]),cm[2][2]/sum(cm[2])))
# fit(x1, ctarg) - precisão de 94%

Voting Classifier


C:\Users\Usuario\Anaconda3\lib\site-packages\sklearn\svm\base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


[[ 721   22   24]
 [  27  675   66]
 [   7   42 1606]]
Accuracy: 0.94, ei: 0.94 , ie: 0.88, n: 0.97


for i in range(len(data)):
    if len(data[i]) != 141:
        print(len(data[i]), 'deu ruim')


In [1]:
import sklearn
print(sklearn.__version__)

0.20.3


In [12]:
a=[0,0,0,1,2,0]
while a[0] not in [1,2]:
    del(a[0])
print(a)

[1, 2, 0]


In [17]:
print(f4[0])

['AF059650  ( 1, 61, 0', 'CATTCTGGATGATGCAGCGAACAATCCTCTCTACTGTGTCCTCATTCATGTCATTCGTCCTCAGCTGAAGGATGAAAGAGATTGGCAAGGAAGCAGGAATGAGTCAGCTTTGGGGGAATATTACCTGGTGGCTCCCATCA\n']


In [3]:
a=[[1,1,1],[2,8,2]]
print(sum(a[0]+a[1]))

15
